src/kinsol/kinsol.c

```diff
@@ -3126,6 +3203,19 @@ static int AndersonAcc(KINMem kin_mem, N_Vector gval, N_Vector fv, N_Vector x,
   retval = N_VDotProdMulti((int)lAA, fv, kin_mem->kin_q_aa, gamma);
   if (retval != KIN_SUCCESS) { return (KIN_VECTOROP_ERR); }
 
+  /* if using adaptive damping, compute the gain factor */
+  sunrealtype gain = SUN_RCONST(-1.0);
+
+  if (kin_mem->kin_damping_fn)
+  {
+    sunrealtype qt_fv_norm = ZERO;
+    for (i = 0; i < lAA; i++) { qt_fv_norm += gamma[i] * gamma[i]; }
+    qt_fv_norm = SUNRsqrt(qt_fv_norm);
+
+    sunrealtype fv_norm = SUNRsqrt(N_VDotProd(fv, fv));
+    gain                = SUNRsqrt(ONE - SUNSQR(qt_fv_norm / fv_norm));
```

Comment:
I think this is equivalent with fewer math operations, but please double check.
```suggestion
    sunrealtype qt_fv_norm_sqr = ZERO;
    for (i = 0; i < lAA; i++) { qt_fv_norm_sqr += gamma[i] * gamma[i]; }

    sunrealtype fv_norm_sqr = N_VDotProd(fv, fv);
    gain                = SUNRsqrt(ONE - qt_fv_norm_sqr / fv_norm_sqr);
```

Reply:
Almost, only one of three square roots in the original version could be removed i.e., this could be replaced with
```
sunrealtype qt_fv_norm_sqr = ZERO;
for (i = 0; i < lAA; i++) { qt_fv_norm_sqr += gamma[i] * gamma[i]; }

sunrealtype fv_norm_sqr = N_VDotProd(fv, fv);
gain                = SUNRsqrt(ONE - SUNRsqrt(qt_fv_norm_sqr / fv_norm_sqr));
```

Reply:
I ended up removing and instead update the `DampingFn` to take the data need to compute the gain as an input so the gain is not computed for damping approaches that don't need it.